documentation/rails.md

```diff
@@ -0,0 +1,124 @@
+```
+
+to require your assets. [Here](https://github.com/infinum/webpack-rails-manifest-plugin/blob/master/example/webpack.config.js#L12) is an example on how to add a digest to a file.
+
+**Take note that you can't use Rails `image_tag`, `stylesheet_link_tag`, `javascript_include_tag` helper.**
+
+#### 2. Monkey patching rails assets helpers
+
+Be sure to remove rails sprockets entirely by editing your `applicaton.rb` file
+
+```Ruby
+# Comment this out
+# require 'rails/all'
+
+# Include every module separately
+require 'rails'
+# Pick the frameworks you want:
+require "active_model/railtie"
+require "active_job/railtie"
+require "active_record/railtie"
+require "action_controller/railtie"
+require "action_mailer/railtie"
+require "action_view/railtie"
+# require "sprockets/railtie"
+require "rails/test_unit/railtie"
+```
+
+And after that copy the code below to your initializers folder `app/config/initializers/webpack-rails-manifest.rb`
+
+```Ruby
+# https://github.com/rails/rails/blob/v4.2.6/actionview/lib/action_view/helpers/asset_tag_helper.rb
+
+module ActionView
+  module Helpers
+    module AssetTagHelper
+      def image_tag(source, options={})
+        options = options.symbolize_keys
+
+        src = options[:src] = "/assets/#{webpack_manifest.fetch(source)}"
+
+        unless src =~ /^(?:cid|data):/ || src.blank?
+          options[:alt] = options.fetch(:alt){ image_alt(src) }
+        end
+
+        options[:width], options[:height] = extract_dimensions(options.delete(:size)) if options[:size]
+        tag("img", options)
+      end
+
+      def javascript_include_tag(*sources)
+        options = sources.extract_options!.stringify_keys
+        path_options = options.extract!('protocol', 'extname').symbolize_keys
+        sources.uniq.map { |source|
+          tag_options = {
+            "src" => "/assets/#{webpack_manifest.fetch(source + '.js')}"
+          }.merge!(options)
+          content_tag(:script, "", tag_options)
+        }.join("\n").html_safe
+      end
+
+      def stylesheet_link_tag(*sources)
+        options = sources.extract_options!.stringify_keys
+        path_options = options.extract!('protocol').symbolize_keys
+
+        sources.uniq.map { |source|
+          tag_options = {
+            "rel" => "stylesheet",
+            "media" => "screen",
+            "href" => "/assets/#{webpack_manifest.fetch(source + '.css')}"
+          }.merge!(options)
+          tag(:link, tag_options)
+        }.join("\n").html_safe
+      end
+
+      private
+
+        def webpack_manifest
+          @manifest ||= JSON.parse(File.read('manifest.json'))
+        rescue
+          fail 'Please run webpack'
+        end
+    end
+  end
+end
+```
+
+Now, you can keep using rails helpers to get the assets
+
+```Ruby
+= image_tag('logo.svg')
+
+= stylesheet_link_tag 'application', media: 'all'
+= javascript_include_tag 'application'
+```
+
+NOTE: this is an example for rails 4.2.6, the code above may vary on different rails versions
```

Comment:
this example **uses** Rails 4.2.6 (although lowercase is also okay, as it's a gem name here)